components/datetime/src/fields/symbols.rs

```diff
@@ -434,36 +478,86 @@ impl LengthType for Month {
-}; Numeric; WeekULE);
-
-field_type!(Weekday; {
-    'E' => Format = 0,
-    'e' => Local = 1,
-    'c' => StandAlone = 2,
-}; WeekdayULE);
+field_type!(
+    /// An enum for the possible symbols of a day field in a date pattern.
+    Day; {
+        /// Field symbol for day of month (numeric).
+        'd' => DayOfMonth = 0,
+        /// Field symbol for day of year (numeric).
+        'D' => DayOfYear = 1,
+        /// Field symbol for the day of week occurrence relative to the month (numeric).
+        ///
+        /// For the example `"2nd Wed in July"`, this field would provide `"2"`.  Should likely be paired with the [`Weekday`] field.
+        'F' => DayOfWeekInMonth = 2,
+        /// Field symbol for the modified Julian day (numeric).
+        ///
+        /// The value of this field differs from the conventional Julian day number in a couple of ways, which are based on measuring relative to the local time zone.
+        'g' => ModifiedJulianDay = 3,
+    };
+    Numeric;
+    DayULE
+);
+
+field_type!(
+    /// An enum for the possible symbols of an hour field in a date pattern.
+    Hour; {
+        /// Field symbol for numeric hour [0-11].
+        'K' => H11 = 0,
+        /// Field symbol for numeric hour [1-12].
+        'h' => H12 = 1,
+        /// Field symbol for numeric hour [0-23].
+        'H' => H23 = 2,
+        /// Field symbol for numeric hour [1-24].
+        'k' => H24 = 3,
+    };
+    Numeric;
+    HourULE
+);
+
+field_type!(
+    /// An enum for the possible symbols of a second field in a date pattern.
+    Second; {
+        /// Field symbol for second (numeric).
+        's' => Second = 0,
+        /// Field symbol for fractional second (numeric).
+        ///
+        /// Truncates, like other numeric time fields, but in this case to the number of digits specified by the field length.
+        'S' => FractionalSecond = 1,
+        /// Field symbol for milliseconds in day (numeric).
+        ///
+        /// This field behaves exactly like a composite of all time-related fields, not including the zone fields.
+        'A' => Millisecond = 2,
+    };
+    Numeric;
+    SecondULE
+);
+
+field_type!(
+    /// An enum for the possible symbols of a week field in a date pattern.
+    Week; {
+        /// Field symbol for week of year (numeric).
+        ///
+        /// When used in a pattern with year, use [`Year::WeekOf`] for the year field instead of [`Year::Calendar`].
+        'w' => WeekOfYear = 0,
+        /// Field symbol for week of month (numeric).
+        'W' => WeekOfMonth = 1,
+    };
+    Numeric;
+    WeekULE
+);
+
+field_type!(
+    /// An enum for the possible symbols of a weekday field in a date pattern.
+    Weekday;  {
+        /// Field symbol for day of week name, format style.
+        'E' => Format = 0,
+        /// Field symbol for local day of week number/name, format style.
+        'e' => Local = 1,
+        /// Field symbol for stand-alone local day of week number/name.
```

Comment:
```suggestion
        /// Field symbol for stand-alone local day of week number/name.
        ///
        /// The stand-alone weekday name is used when the weekday is displayed by itself. This may differ from the standard form based on the language and context.
```